node/pkg/devnet/deterministic_bridge_key.go

```diff
@@ -3,18 +3,36 @@ package devnet
 import (
 	"crypto/ecdsa"
 	"crypto/elliptic"
-	mathrand "math/rand"
+
+	eth_crypto "github.com/ethereum/go-ethereum/crypto"
 )
 
 // InsecureDeterministicEcdsaKeyByIndex generates a deterministic ecdsa.PrivateKey from a given index.
 func InsecureDeterministicEcdsaKeyByIndex(c elliptic.Curve, idx uint64) *ecdsa.PrivateKey {
-	// use 555 as offset to deterministically generate key 0 to match vaa-test such that
-	// we generate the same key.
-	r := mathrand.New(mathrand.NewSource(int64(555 + idx))) //#nosec G404 Testnet/devnet keys are not secret.
-	key, err := ecdsa.GenerateKey(c, r)
+
+	// with golang <= 1.19, we used the following code to generate deterministic keys.
+	// But in golang 1.20, ecdsa.GenerateKey became non-deterministic and therefore the keys are now hardcoded.
```

Comment:
Why change this?

Reply:
Not sure what you're referring to by "this", but as the code comment says this function needed to be rewritten because this function needs to produce deterministic keys, as enforced in https://github.com/wormhole-foundation/wormhole/blob/7d0d92997e1f7fe9d95a0850239e12081b087e1b/node/pkg/devnet/deterministic_bridge_key_test.go but with go 1.20 ecdsa.GenerateKey became non-deterministic